cgsmiles/resolve.py

```diff
@@ -210,25 +246,44 @@ def squash_atoms(self):
             self.molecule.nodes[node_to_keep]['fragid'] += self.molecule.nodes[node_to_keep]['contraction'][node_to_remove]['fragid']
 
     def resolve(self):
+        """
+        Resolve a CGSmiles string once and return the next resolution.
+        """
+        # check if this is an all-atom level resolution
+        if self.resolution_counter == self.resolutions -1 and self.last_all_atom:
+            all_atom = True
+        else:
+            all_atom = False
+
+        # get the next set of fragments
+        if self.fragment_strs:
+            fragment_str = self.fragment_strs[self.resolution_counter]
+            # read the fragment str and populate dict
+            self.fragment_dict = read_fragments(fragment_str, all_atom=all_atom)
+        else:
+            self.fragment_dict = self.fragment_dicts[self.resolution_counter ]
```

Comment:
Since the user can pass a `fragment_dicts` to `__init__` I would suggest fully consuming `fragment_strs` in `__init__` to populate `fragment_dicts` instead. Then the strs don't need to be an attribute any longer.

Reply:
when refactoring I already had the same thought; it makes the most sense